packages/components/pie-card-container/src/index.ts

```diff
@@ -33,31 +38,34 @@ export class PieCardContainer extends LitElement implements CardContainerProps {
 
     render () {
         const {
+            tag,
+            disabled,
+            variant,
             href,
             target,
             rel,
-            disabled,
-            variant,
         } = this;
 
+        const isButton = tag === 'button';
+        const element = unsafeStatic(isButton ? 'div' : tag);
+
         return html`
-                <div
+                <${element}  
```

Comment:
@raouf I don't think we should use a button here. It goes against the HTML spec of how a button should be used.

Note here: https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element

The button element should only contain _phrasing content_. Our slots will allow users to add (perhaps quite often) Flow content. 

Happy to discuss further offline :)

Reply:
Hi @jamieomaguire 
good point. we aren't using a div with the needed attributes to make sure it behaves as a button as mentioned in the ticket. 

```
const element = unsafeStatic(isButton ? 'div' : tag);
```

Reply:
🤦 I need my morning coffee. Sorry @raoufswe!

I do have another comment - it seems that this kind of dynamic element interpolation breaks not only syntax highlighting but potentially Lit binding/ts validation. Please see this example. Note what value I have provided for href in the conditional.

<img width="1683" alt="Screenshot 2023-09-12 at 08 55 32" src="https://github.com/justeattakeaway/pie/assets/16766185/132c8a3f-179e-4a93-a6a8-4e53744e73f7">

Reply:
I know its long winded but please can we just render the two HTML variants separately in an if else? It will preserve highlighting etc even though its a bit long winded

Reply:
updated to render different templates based on the interaction type 💯